File: Sorting method/prefix sort.py
n = int(input())
a = [0] * 100000

# Input and count occurrences of each element
for _ in range(n):
    x = int(input())
    a[x] += 1

# Print the sorted array
for i in range(100000):
    while a[i] != 0:
        print(i)
        a[i] -= 1

File: Sorting method/prefix sort.java
import java.util.Scanner;

public class CountingSort {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] a = new int[100000];

        // Input and count occurrences of each element
        for (int i = 0; i < n; i++) {
            int x = scanner.nextInt();
            a[x]++;
        }

        // Print the sorted array
        for (int i = 0; i < 100000; i++) {
            while (a[i] != 0) {
                System.out.println(i);
                a[i]--;
            }
        }
    }
}
